src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -582,11 +582,33 @@ private static TDeclarationNode AddStatementsToBaseMethodDeclaration<TDeclaratio
         var finalMember = baseMethodDeclaration
             .WithExpressionBody(null)
             .WithSemicolonToken(default)
-            .WithBody(body.WithStatements(body.Statements.AddRange(StatementGenerator.GenerateStatements(statements))));
+            .WithBody(AddStatementsToBlock(body, statements));
 
         return Cast<TDeclarationNode>(finalMember);
     }
 
+    public static BlockSyntax AddStatementsToBlock(BlockSyntax block, IEnumerable<SyntaxNode> statements)
+    {
+        var statementsArray = StatementGenerator.GenerateStatements(statements);
+        if (statementsArray.Count > 0)
```

Comment:
>if (statementsArray.Count > 0) [](http://example.com/codeflow?start=8&length=30)

nit: could simplify the empty statement case to

if (statementsArray.Count == 0)
    return block;